wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceService.java

```diff
@@ -664,11 +669,21 @@ public void deleteProject(
   @ApiOperation(value = "Get workspace server configuration values")
   @ApiResponses({@ApiResponse(code = 200, message = "The response contains server settings")})
   public Map<String, String> getSettings() {
-    return ImmutableMap.of(
-        Constants.SUPPORTED_RECIPE_TYPES,
-        Joiner.on(",").join(workspaceManager.getSupportedRecipes()),
-        CHE_WORKSPACE_AUTO_START,
-        Boolean.toString(cheWorkspaceAutoStart));
+    if (pluginRegistryUrl == null) {
```

Comment:
I think it would be better to rewrite this method with using Map builder.
```java
    Builder<String, String> settings = ImmutableMap.builder();

    settings.put(Constants.SUPPORTED_RECIPE_TYPES, Joiner.on(",").join(workspaceManager.getSupportedRecipes()));
    settings.put(CHE_WORKSPACE_AUTO_START, Boolean.toString(cheWorkspaceAutoStart));

    if (pluginRegistryUrl != null) {
      settings.put(CHE_WORKSPACE_PLUGIN_REGISTRY_ULR, pluginRegistryUrl);
    }

    return settings.build();
```